Effisense/sensor-dashboard
# Setup `packages/timeseries`, and integrate with TimescaleDB

Closes: #15

## Checklist


## Changelog

- Setup Knex for building queries
- Setup Kanel for autogenerating types, given a PostgreSQL database schema
- Setup Turborepo targets, such that you can run `pnpm gen:types`  in root and generate the types from TimescaleDB
- Initialized connection to the database, and exposed a variables `Sensor` to be used when querying the database

## How to test

- Ensure you have up-to-date environment variables. Visit Vault and compare with the contents of `.env.example`.
- Ensure all dependencies are installed (`pnpm i`)
- Generate types (`pnpm gen:types`) in `sensor-dashboard`
- Observe that types from TimescaleDB was autogenerated in `packages/timeseries/src/schemas`
- Observe that we have a `Sensor` variable in `index.ts` in this package, in order to facilitate building queries

This `Sensor` variable will be used in the website by running for instance `const result = await Sensor.select('something', 'something_else').where(...)`. We get type safety on the parameter we put in, and on the result we get out.
